### PR TITLE
ebirdhistoric

### PR DESCRIPTION
**Pull request by gbabineau**
@sebpardo, no hurry on this. I was looking to figure out a way to do something, then thought I might need this function to look at historical observations but then didn't need it after all. But since I had it 80% working, I finished it and added the tests. Since I was able to use ebird_GET, the implementation was straightforward. Since there were so many optional parameters, the only slow part was adding tests for them all.

What I really wanted to do was go after some breeding code review data that does not seem to be accessible via this interface.

### CODE REVIEW

File: R/ebirdhistoricobs.R [hunk 1]
```diff
@@ -0,0 +1,93 @@
+#' Historic observations on a date at a region or set of hotspots
+#' 
+#' Returns sighting information reported in a given region or set of hotspots
+#' 
+#' @param loc (required) Region code or locID (for hotspots). Region code can
+#'    be country code (e.g. "US"), subnational1 (states/provinces, e.g. "US-NV"), or
+#'    subnational2 code (counties, e.g. "US-VA-003").
+#' @param date (required) Date of historic observation date formatted according 
+#'    to ISO 8601 (e.g. 'YYYY-MM-DD', or 'YYYY-MM-DD hh:mm'). Hours and minutes 
+#'    are excluded
```
Comment (sebpardo) [inline R/ebirdhistoricobs.R]: Make sure you're consistent with punctuation in the documentation.

Comment (sebpardo) [reply R/ebirdhistoricobs.R]: Actually, I can make edits on the documentation once the PR is ready.

File: R/ebirdhistoricobs.R [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  max = 10000, fieldSet='simple', provisional = FALSE, limitToHotspots = FALSE, sleep=0, 
+  key=NULL, ...)
+{
+
+
+  historicDate=as.Date(date)
+  if (Sys.Date()<historicDate) {
+    stop(paste0("date must be in the past"))
+  }
+  if (historicDate<'1900-01-01'){
```
Comment (sebpardo) [inline R/ebirdhistoricobs.R]: There is eBird data that is from the late 1800s, and the API allows data from the year 1800 onwards. Make sure it matches with what eBird allows.

File: R/ebirdhistoricobs.R [hunk 3]
```diff
@@ -0,0 +1,93 @@
+#' Historic observations on a date at a region or set of hotspots
```
Comment (sebpardo) [inline R/ebirdhistoricobs.R]: I would rename the function `ebirdhistoric()` as it's reads much better and pretty much other function also deals with observations implicitly.

**Comment by sebpardo** [comment]
Thanks Guy, looks great! Added a few comments. Don't worry about the GitHub tests failing, as Travis-CI doesn't load the API keys for PR tests, and there's something up with the loading of `stringi` on Appveyor. Just double check that `devtools::test()` runs with no errors in your machine.

File: R/ebirdhistoricobs.R (hunk 1 above)
Comment (gbabineau) [reply R/ebirdhistoricobs.R]: Ok. I made it more consistent. In many cases, I copied the parameter descriptions from other files. So they are probably not consistent in the other files. It should be consistent though.

File: R/ebirdhistoricobs.R (hunk 3 above)
Comment (gbabineau) [reply R/ebirdhistoricobs.R]: I made it ebirdhistorical. In thinking about it, they are more historical than historic.

File: R/ebirdhistoricobs.R (hunk 2 above)
Comment (gbabineau) [reply R/ebirdhistoricobs.R]: Ok. I saw 1900 somewhere else and thought that was the date. I changed it to 1800.

**Comment by gbabineau** [comment]
Thanks - I made most of the changes. I'll have to figure out the rebase. The tests all pass. I changed the name slightly differently than you suggested. See what you think.

**Comment by gbabineau** [comment]
I failed on the rebase it looks like! Unless you see a way to fix, I am thinking just close the pull request and redo.

**Comment by sebpardo** [comment]
No worries. Probably easiest to do what you suggested and start a new PR.

**Comment by gbabineau** [comment]
closing to create a clean one

**Pull request closed without merging**